cli/azd/cmd/extensions.go

```diff
@@ -0,0 +1,174 @@
+		Use:                extension.Name,
+		Short:              extension.Description,
+		Long:               extension.Description,
+		DisableFlagParsing: true,
+	}
+
+	cmd.SetHelpFunc(func(c *cobra.Command, s []string) {
+		_ = serviceLocator.Invoke(invokeExtensionHelp)
+	})
+
+	root.Add(extension.Name, &actions.ActionDescriptorOptions{
+		Command:        cmd,
+		ActionResolver: newExtensionAction,
+		GroupingOptions: actions.CommandGroupOptions{
+			RootLevelHelp: actions.CmdGroupExtensions,
+		},
+	})
+
+	return nil
+}
+
+// invokeExtensionHelp invokes the help for the extension
+func invokeExtensionHelp(console input.Console, commandRunner exec.CommandRunner, extensionManager *extensions.Manager) {
+	extensionName := os.Args[1]
+	extension, err := extensionManager.GetInstalled(extensionName)
+	if err != nil {
+		fmt.Println("Failed running help")
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Println("Failed running help")
+	}
+
+	extensionPath := filepath.Join(homeDir, extension.Path)
+
+	runArgs := exec.
+		NewRunArgs(extensionPath, os.Args[2:]...).
+		WithStdIn(console.Handles().Stdin).
+		WithStdOut(console.Handles().Stdout).
+		WithStdErr(console.Handles().Stderr)
+
+	_, err = commandRunner.Run(context.Background(), runArgs)
+	if err != nil {
+		fmt.Println("Failed running help")
+	}
+}
+
+type extensionAction struct {
+	console          input.Console
+	commandRunner    exec.CommandRunner
+	lazyEnv          *lazy.Lazy[*environment.Environment]
+	extensionManager *extensions.Manager
+	cmd              *cobra.Command
+	args             []string
+}
+
+func newExtensionAction(
+	console input.Console,
+	commandRunner exec.CommandRunner,
+	lazyEnv *lazy.Lazy[*environment.Environment],
+	extensionManager *extensions.Manager,
+	cmd *cobra.Command,
+	args []string,
+) actions.Action {
+	return &extensionAction{
+		console:          console,
+		commandRunner:    commandRunner,
+		lazyEnv:          lazyEnv,
+		extensionManager: extensionManager,
+		cmd:              cmd,
+		args:             args,
+	}
+}
+
+func (a *extensionAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	extensionName := a.cmd.Use
+
+	extension, err := a.extensionManager.GetInstalled(extensionName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get extension %s: %w", extensionName, err)
+	}
+
+	allEnv := []string{}
+	allEnv = append(allEnv, os.Environ()...)
+
+	forceColor := !color.NoColor
+	if forceColor {
+		allEnv = append(allEnv, "FORCE_COLOR=1")
+	}
+
+	env, err := a.lazyEnv.GetValue()
+	if err == nil && env != nil {
+		allEnv = append(allEnv, env.Environ()...)
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current working directory: %w", err)
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user home directory: %w", err)
+	}
+
+	extensionPath := filepath.Join(homeDir, extension.Path)
+
+	_, err = os.Stat(extensionPath)
```

Comment:
Would it be possible to move this logic in to `GetInstalled`?  Could the `extension` returned have a full path (including the home directory) and do the `Stat` to ensure the binary is there?